pkg/api/mock/matcher_object_tags.go

```diff
@@ -0,0 +1,40 @@
+package mock
+
+import (
+	"fmt"
+
+	gomegaTypes "github.com/onsi/gomega/types"
+)
+
+// ObjectTaggedWith checks if mockAPIObject is tagged with tags
+func ObjectTaggedWith(tags ...string) gomegaTypes.GomegaMatcher {
+	return &objectTaggedWith{
+		tags: tags,
+	}
+}
+
+type objectTaggedWith struct {
+	tags []string
+}
+
+func (m *objectTaggedWith) Match(actual interface{}) (bool, error) {
+	obj, ok := actual.(*mockAPIObject)
+	if !ok {
+		return false, fmt.Errorf("ObjectTaggedWith matcher expects an *mockAPIObject")
+	}
+
+	for _, tag := range m.tags {
```

Comment:
can you implement this with `ContainElements()`, using the `(Negated)FailureMessage` from that? Would give us a list of which tags are missing or wrongly set.

Reply:
I've missed that part. I think we only need this in `FailureMessage` as `NegatedFailureMessage` is only displayed when $\text{searched tags} \subset \text{object tags}$ and it's negated.